openwisp_firmware_upgrader/base/models.py

```diff
@@ -205,6 +206,11 @@ def _remove_file(self):
             msg = 'firmware image not found while deleting {0}:\n{1}'
             logger.error(msg.format(self, path))
 
+    def _remove_empty_directory(self):
+        path = os.path.dirname(self.file.path)
+        if not os.listdir(path):
```

Comment:
could you double check that path is not equal to `settings.MEDIA_ROOT`? Just to be sure we won't accidentally delete that, since we used it previously (I'd add a comment with this info and a TODO: remove this in first release)